lib/dynamic-cache.js

```diff
@@ -35,15 +37,31 @@ DynamicCache.prototype.update = function (cb) {
       this._queue = [];
     }
   }.bind(this);
+
   var getDeps = function (filename) {
-    if (this._files[filename]) {
-      return [[filename]].concat(this._files[filename].map(getDeps)).reduce(function (a, b) {
```

Comment:
Yeah that was definitely one problem.
But the whole map, reduce, concat made it also pretty slow.
My version is much faster in our setup